File: DatabaseMicroservice/src/main/java/com/licenta/databasemicroservice/persistence/repository/JobRepository.java
package com.licenta.databasemicroservice.persistence.repository;

import com.licenta.databasemicroservice.persistence.entity.Job;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Set;

public interface JobRepository extends JpaRepository<Job, Long> {
    Set<Job> findJobsByCompanyIndustryId(Long companyIndustryId);
}

File: FilterMicroservice/src/main/kotlin/com/licenta/searchmicroservice/persistence/repository/JobRepository.kt
package com.licenta.searchmicroservice.persistence.repository

import com.licenta.searchmicroservice.persistence.entity.Job
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface JobRepository: JpaRepository<Job, String> {

    fun findAllByQueryId(queryId: String, pageable: Pageable): Page<Job>
    fun deleteAllByQueryId(queryId: String);
}